[8.0] Improve + modularize generation of UBL and CII XML

alexis-via opened the pull request:
This PR is required for my implementation of Chorus: https://github.com/OCA/l10n-france/pull/103

alexis-via committed:
Improve + modularize generation of UBL and CII XML

hbrunn reviewed (changes requested):
looks fine technically, but I think you should add tests too

alexis-via commented:
I added tests in the upper-level module l10n_fr_chorus_ubl which has inherits of the new method I implemented in this PR, which allows me to test that the "full" XML is properly generated.

hbrunn commented:
and there's no chance to test something in the general modules? I'm very unfamiliar with UBL, that's why I can't judge this

alexis-via committed:
Add script mass_invoice_import.py

florentboucherieconsulting reviewed (approved):
Work fine : it imported more than 300 pdf invoice for me !

alexis-via committed:
FIX typo in comment

thomaspaulb commented on base_ubl/models/ubl.py:
```diff
@@ -124,21 +186,11 @@ def _ubl_add_party(
-            tax_scheme_id.text = 'VAT'
+        self._ubl_add_party_tax_scheme(
+            commercial_partner, party, ns, version=version)
+        if company:
```
@alexis-via Do you remember if there was any particular reason why the PartyLegalEntity is only added for the partner who has the `res.company` in Odoo, eg. the sender of a sales invoice? In Netherlands the PartyLegalEntity is also required for the customer side.

alexis-via replied:
I don't think there is a good reason. I think it was linked to the fact that i decided to use the commercial_partner of the invoice as invoice address, which is probably a bad reason, cf https://github.com/OCA/edi/issues/104 As I said in this issue, I'm OK to modify that.

thomaspaulb replied:
@alexis-via Thanks for pointing me to the issue! I saw the fix for it was even already merged to the main 10.0 branch, so I backported it to the version I needed it for.
